parsers/powershell.c

```diff
@@ -506,6 +529,38 @@ static bool parseFunction (tokenInfo *const token)
 	return readNext;
 }
 
+/* parse a class
+ *
+ * 	class MyClass {}
+ */
+static bool parseClass (tokenInfo *const token)
+{
+	bool readNext = true;
+	tokenInfo *nameFree = NULL;
+	const char *access;
+
+	readToken (token);
+
+	if (token->type != TOKEN_IDENTIFIER)
+		return false;
+
+	nameFree = newToken ();
+	copyToken (nameFree, token, true);
+	readToken (token);
+
+	makeClassTag (nameFree);
+
+	if (token->type == TOKEN_OPEN_CURLY)
+		enterScope (token, nameFree->string, K_CLASS);
+	else
+		readNext = false;
+
+	if (nameFree)
```

Comment:
Can nameFree be NULL?
If yes, nameFree->string at line 554 is unsafe.

Reply:
I found the same code pattern in parseFunction().
If you are intrested in fixing parseFunction(), you can do it in this pull request but please make a separate commit.

Reply:
I assume that nameFree has a value and that nameFree is not null.
I would remove the `if` line.